apps/example-orchestrator-ui/components/Subscriptions/Subscriptions.tsx

```diff
@@ -149,63 +172,49 @@ export const Subscriptions: FC<SubscriptionsProps> = ({
-        'description',
-        'status',
-        'insync',
-        'organisationName',
-        'organisationAbbreviation',
-        'productName',
-        'tag',
-        'startDate',
-        'endDate',
-        'note',
-    ];
-
-    const leadingControlColumns: ControlColumn<Subscription>[] = [
-        {
-            id: 'inlineSubscriptionDetails',
-            width: 40,
-            rowCellRender: () => (
-                <EuiFlexItem>
-                    <PlusCircleFill color={theme.colors.mediumShade} />
-                </EuiFlexItem>
-            ),
-        },
-    ];
-
     return (
-        <Table
-            data={mapApiResponseToSubscriptionTableData(data)}
-            pagination={{
-                pageSize: pageSize,
-                pageIndex: Math.floor(pageIndex / pageSize),
-                pageSizeOptions: [5, 10, 15, 20, 25, 100],
-                totalRecords: parseInt(data.subscriptions.pageInfo.totalItems),
-                onChangePage: (updatedPageNumber) =>
-                    setPageIndex(updatedPageNumber * pageSize),
-                onChangeItemsPerPage: (itemsPerPage) =>
-                    setPageSize(itemsPerPage),
-            }}
-            columns={tableColumnConfig}
-            leadingControlColumns={leadingControlColumns}
-            initialColumnOrder={initialColumnOrder}
-            dataSorting={{
-                columnId: sortedColumnId,
-                sortDirection: sortOrder.order,
-            }}
-            updateDataSorting={(dataSorting) =>
-                setSortOrder({
-                    field: dataSorting.columnId,
-                    order: dataSorting.sortDirection,
-                })
-            }
-        ></Table>
+        <>
+            <Table
+                data={mapApiResponseToSubscriptionTableData(data)}
+                columns={tableColumnsWithExtraNonDataFields}
+                hiddenColumns={hiddenColumns}
+                dataSorting={{
+                    columnId: sortedColumnId,
+                    sortDirection: sortOrder.order,
+                }}
+                onDataSort={(newSortColumnId) =>
+                    setSortOrder({
+                        field: newSortColumnId,
+                        order: determineNewSortOrder(
+                            sortedColumnId,
+                            sortOrder.order,
+                            newSortColumnId,
+                        ),
+                    })
+                }
+                pagination={{
+                    pageSize: pageSize,
+                    pageIndex: Math.floor(pageIndex / pageSize),
+                    pageSizeOptions: [5, 10, 15, 20, 25, 100],
+                    totalItemCount: parseInt(
+                        data.subscriptions.pageInfo.totalItems,
+                    ),
+                }}
+                isLoading={isFetching}
+                onCriteriaChange={(criteria) => {
+                    const { page } = criteria;
```

Comment:
Destructuring within destructuring might be less readable, therefore I destructure the page from criteria first. And inside the if-statement I destructure index and size from page.

Reply:
- I must stand my point
- `Destructuring` is not about readability. It's a special syntax to unpack arrays or objects into a bunch of variables when convenient. 
- `Destructuring` when accessing not repeating variable is not necessarily required. `Destructuring` twice instead of once is unnecessary.
- Less code improves readability and vice versa.